### ISSUE TITLE
Lag typing large file with Matlab extention

### ISSUE THREAD

**Issue by pjacquet88** [Gimly/vscode-matlab#162]
Issue Type: <b>Performance Issue</b>

On large matlab files >800 lines,
I have several seconds of lag when typing a letter or even when I hit the arrow to move inside the code. Suprisingly backspace or ctrl+z works intantly. I do not have this latency when disable matlab extention but also my emacs extension. Could it be a compatibility issue ?

Extension version: 2.3.1
VS Code version: Code 1.62.2 (3a6960b964327f0e3882ce18fcebd07ed191b316, 2021-11-11T20:56:38.428Z)
OS version: Windows_NT x64 10.0.18363
Restricted Mode: No
Fetching remote diagnostics for 'ssh-remote+7b22686f73744e616d65223a22746c6c69663230222c2275736572223a226a61637175657031227d' timed out.
Remote OS version: Linux x64 3.10.0-693.el7.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz (8 x 2419)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.69GB (6.78GB free)|
|Process Argv|--crash-reporter-id 45a3798f-1e4c-4fd3-aa26-0d04032a23df|
|Screen Reader|no|
|VM|0%|

Fetching remote diagnostics for 'ssh-remote+7b22686f73744e616d65223a22746c6c69663230222c2275736572223a226a61637175657031227d' timed out.

|Item|Value|
|---|---|
|Remote|SSH: tllif20|
|OS|Linux x64 3.10.0-693.el7.x86_64|
|CPUs|Intel(R) Xeon(R) Gold 6136 CPU @ 3.00GHz (24 x 3000)|
|Memory (System)|376.40GB (313.87GB free)|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	    96	 18616	code main
    0	    65	  1756	   shared-process
    0	    37	 14076	     ptyHost
    0	     6	 13240	       C:\WINDOWS\System32\cmd.exe /c (type "C:\Users\jacquep1\AppData\Local\Temp\vscode-linux-multi-line-command-tllif20-406830708.sh" | ssh -T -D 56084 "jacquep1@tllif20" bash) & exit /b 0
    0	     8	  6512	         ssh  -T -D 56084 "jacquep1@tllif20" bash
    0	     5	 13816	       console-window-host (Windows internal process)
    0	     5	 17256	       C:\WINDOWS\System32\cmd.exe /c (type "C:\Users\jacquep1\AppData\Local\Temp\vscode-linux-multi-line-command-tllif20-319232353.sh" | ssh -T -D 59838 "jacquep1@tllif20" bash) & exit /b 0
    0	     7	 11212	         ssh  -T -D 59838 "jacquep1@tllif20" bash
    0	     5	 22136	       console-window-host (Windows internal process)
    0	     9	  6544	   crashpad-handler
    1	    53	  7672	   window (Issue Reporter)
    0	   128	  8372	   window (job_shop.m - test [SSH: tllif20] - Visual Studio Code)
    0	    35	 14896	     extensionHost
    0	    25	 11040	   utility
    0	    16	 12080	   utility
    0	   180	 13808	   gpu-process
    0	   186	 22232	   window (gurobi_model_oneband_allts.m - 52_idem49_seuilincline245 [SSH: tllif20] - Visual Studio Code)
    0	    59	  7420	     extensionHost
Fetching remote diagnostics for 'ssh-remote+7b22686f73744e616d65223a22746c6c69663230222c2275736572223a226a61637175657031227d' timed out.

Remote: SSH: tllif20
```

</details>
<details>
<summary>Workspace Info</summary>

```

Fetching remote diagnostics for 'ssh-remote+7b22686f73744e616d65223a22746c6c69663230222c2275736572223a226a61637175657031227d' timed out.;
```

</details>
<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30429732
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
wslgetstarted:30449410
dsvsc009:30452663
pythonvspyt640:30450904
vscscmwlcmc:30438804
vscgsvidc:30447482
vscaat:30438848
pynewfile477cf:30450039

```

</details>

**Comment by SNDST00M** [comment]
It's probably #157. But let's address it and see if your issue goes away!

**Comment by M00TSDNS** [comment]
Workaround for now:

```shell
$ cd ~/.vscode/extensions/gimly81.matlab
$ npm install vscode-textmate-languageservice@^latest
```

**Comment by M00TSDNS** [comment]
Fix to be released SOON™, subscribe to #171.
